src/lib/moment/year_guess.js

```diff
@@ -0,0 +1,61 @@
+/*
+"I'd like to add some ability to specify the year before parsing happens, like
+moment.year(2021).parse('Mon Dec 06, 9:30am CST', 'ddd MMM DD, h:mma')
+Or the moment can try to auto-guess the year by trying 2023, 
+then 2022, and so on (I don't know how far into the past it 
+makes sense to travel, but I guess there can be some limit)"
+
+Make a function that provided a date and format provide the
+last valid year in which the date occured. For example, 2023 
+does not have a December 6th that lands on a Monday. The last
+date in which December 6th landed on a Monday was in 2021.
+Also take into account a 'year' variable that starts backtracking
+from a given year.
+
+The date variable is the date in question.
+The format variable is the format of the date.
+The year variable is the upper bound of years to check
+which is defaulted to the current year.
+
+The output is currently simply the proper moment object.
+*/
+
+export function yearguess(date, format, year=moment().year()){
```

Comment:
Shouldn't the function name be in camelCase?

```suggestion
export function yearGuess(date, format, year=moment().year()){
```